05/ApiHandler.js

```diff
@@ -0,0 +1,35 @@
+class ApiHandler {
+	constructor() {
+		this.apiUrl = 'http://localhost:3000';
+	}
+	load(resource) {
+		return this._fetch(resource);
+	}
+
+	create(resource, userData) {
+		const options = {
+			method: 'POST',
+			body: JSON.stringify(userData),
+			headers: { 'Content-Type': 'application/json' },
+		};
+		return this._fetch(resource, options);
+	}
+
+	_fetch(resource = '', options = { method: 'GET' }) {
+		const path = this.apiUrl + resource;
+		const promise = fetch(path, options);
+
+		return promise
+			.then(resp => {
+				if (resp.ok) {
+					return resp.json();
+				}
+				return Promise.reject(resp);
+			})
+			// dlaczego ten console log psuje mi wysylanie danych do app.js skoro w górnym then juz retrunuje resp.json, to dlaczego console.log moze mi psuc cala logike?!
+			// .then(data => console.log(data))
```

Comment:
Dlatego, że odbierasz dane i juz ich nie zwracasz (więc kolejny `.then` nie ma czego odebrać).
Jak chcesz w ten sposób to powinieneś zrobić tak: 
```
.then(data => {
    console.log(data);
    return data;
})
```

Reply:
ale to nie jest tak ze pierwszy .then w ktorym sprawdzam if(resp.ok) on ma return resp.json, wiec w tym momencie zwracam(return) dane do mojego pliku app,js, wiec kolejne theny nie powinny miec wplywu na dane(ktore juz trafily do mnie za pomoca returna)?!
bo pierwszy then zwrocil mi juz retrunem resp?